teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -80,6 +80,7 @@ same namespace create resource. Related issue
 
 1. Custom task controllers are to be developed by other parties. Custom task
  specification validation by Tektoncd/Pipeline webhooks.
+2. Handling `metadata` for embedded custom tasks i.e. `taskSpec`.
```

Comment:
I do not understand this non-goal here. Please explain a little more.

Also, we already have support for metadata while embedding the task specifications which must be propagated to `Run` or `TaskRun`:

```
type EmbeddedTask struct {
	// +optional
	Metadata PipelineTaskMetadata `json:"metadata,omitempty"`

	// TaskSpec is a specification of a task
	TaskSpec `json:",inline,omitempty"`
}
```

Reply:
Thank you for the question, @pritidesai. *I am sorry*, metadata is supported but it is limited to the sub-fields `Labels` and `annotations`

```go
type PipelineTaskMetadata struct {
	// +optional
	Labels map[string]string `json:"labels,omitempty"`

	// +optional
	Annotations map[string]string `json:"annotations,omitempty"`
}
```

In the normal creation of a custom task resource object, e.g.:
```yaml
apiVersion: custom.tekton.dev/v1alpha1
kind: TaskLoop
metadata:  # A metadata field e.g. name is supported 
  name: simpletaskloop
spec:
  # Task to run (inline taskSpec also works)
  taskRef:
    name: simpletask
  # Parameter that contains the values to iterate
  iterateParam: word
  # Timeout (defaults to global default timeout, usually 1h00m; use "0" for no timeout)
  timeout: 60s
  # Retries for task failure
  retries: 2
```

Once we embed a custom task, specifying it's name becomes irrelevant. But, still is it worth noting that we do not support all the metadata subfields?

I can update the non goal with above clarification, or completely remove it if it leads to confusion by stating the obvious.